### PR TITLE
Expose suppressed and original prompt modes

### LINKED ISSUE #1449: Expose original and suppressed prompt modes on validated authorize requests
**Which version of Duende IdentityServer are you using?**
6.2.1

**Which version of .NET are you using?**
7.0

**Describe the request**

We have an implementation of `ICustomAuthorizeRequestValidator` which contains some logic depending on PromptModes. 

Currently we use a workaround similar to the one introduced with https://github.com/DuendeSoftware/IdentityServer/pull/65. We are looking to remove our workaround and rely on the fix in IdentityServer. This works fine when accessing the promptModes from the returned result of `IIdentityServerInteractionService.GetAuthorizationContextAsync`, but the same is not true for the context passed into `ICustomAuthorizeRequestValidator`, as seen [here](https://github.com/DuendeSoftware/IdentityServer/blob/df2b00bb905905be27a25a3c00397c5020165ef6/src/IdentityServer/Validation/Default/AuthorizeRequestValidator.cs#L120C1-L132C55). `ValidateOptionalParametersAsync` mutates the PromptModes list just before passing it to the custom validator.  

The `ValidatedAuthorizeRequest` does have an `OriginalPromptModes`, but it is internal. Is it possible to make the promptModes available somehow? 

**To Reproduce**

Same as in https://github.com/IdentityServer/IdentityServer4/issues/5002, except instead of trying to access promptModes on the result of `IIdentityServerInteractionService.GetAuthorizationContextAsync`, try and access the prompmodes from within a `ICustomAuthorizeRequestValidator`.

### PR DESCRIPTION
**Pull request by josephdecock**
Closes #1449

### CODE REVIEW

File: src/IdentityServer/Validation/Models/ValidatedAuthorizeRequest.cs [hunk 1]
```diff
@@ -138,26 +138,66 @@ public class ValidatedAuthorizeRequest : ValidatedRequest
...
+    /// <list type="bullet">
+    /// <item><seealso cref="PromptModes"/></item>
+    /// <item><seealso cref="OriginalPromptModes"/></item>
+    /// </list>
+    /// </para>
     /// <value>
     /// The collection of suppressed prompt modes.
     /// </value>
-    internal IEnumerable<string> SuppressedPromptModes { get; set; } = Enumerable.Empty<string>();
+    public IEnumerable<string> SuppressedPromptModes { get; set; } = Enumerable.Empty<string>();
```
Comment (brockallen) [inline src/IdentityServer/Validation/Models/ValidatedAuthorizeRequest.cs:200]: Maybe we want a better name for these, like "ProcessedPromptModes"? Thoughts? Suggestions?

File: src/IdentityServer/Constants.cs [hunk 2]
```diff
@@ -115,7 +115,7 @@ public static class SigningAlgorithms
         //OidcConstants.PromptModes.Create, 
     };
 
-    public const string SuppressedPrompt = "suppressed_" + OidcConstants.AuthorizeRequest.Prompt;
+    public const string ProcessedPrompt = "processed_" + OidcConstants.AuthorizeRequest.Prompt;
```
Comment (josephdecock) [inline src/IdentityServer/Constants.cs:118]: Renaming here makes the names consistent, but is a (very minor) breaking change.

Comment (brockallen) [reply src/IdentityServer/Constants.cs:118]: This would only break someone in-flight, right?

Comment (josephdecock) [reply src/IdentityServer/Constants.cs:118]: Yeah, if you're using the constant or looking at the parameters raw and using the old name, this means you need to make an update.

File: src/IdentityServer/Validation/Models/ValidatedAuthorizeRequest.cs (hunk 1 above)
Comment (brockallen) [reply src/IdentityServer/Validation/Models/ValidatedAuthorizeRequest.cs]: Ok, let's revert the constant and keep the new "Processed" name on the new prop, please.

File: src/IdentityServer/Constants.cs (hunk 2 above)
Comment (brockallen) [reply src/IdentityServer/Constants.cs:118]: Leave the old constant and deprecate it w/ obsolete, please.

**Comment by josephdecock** [comment]
@brockallen this is ready for review again.

File: src/IdentityServer/Constants.cs [hunk 3]
```diff
@@ -115,7 +115,21 @@ public static class SigningAlgorithms
...
     };
 
-    public const string ProcessedPrompt = "processed_" + OidcConstants.AuthorizeRequest.Prompt;
+    /// <summary>
+    /// The name of the parameter passed to the authorize callback to indicate
+    /// prompt modes that have already been used. This constant is deprecated in
+    /// favor of <see cref="ProcessedPrompt"/>.
+    /// </summary>
+    [Obsolete("Use the ProcessedPrompt constant instead.")]
+    public const string SuppressedPrompt = "suppressed" + OidcConstants.AuthorizeRequest.Prompt;
```
Comment (brockallen) [inline src/IdentityServer/Constants.cs:124]: we lost the "_", no?

Comment (josephdecock) [reply src/IdentityServer/Constants.cs:124]: Thanks for catching this - fixed!

**Review by brockallen — APPROVED** [review]

**Comment by brockallen** [comment]
@josephdecock darn, now some conflicts

**Pull request merged**